app/controllers/posts_controller.rb

```diff
@@ -466,6 +466,22 @@ def hide_revision
     render body: nil
   end
 
+  def purge_revisions
+    guardian.ensure_can_purge_post_revisions!
+
+    post = find_post_from_params
+    raise Discourse::InvalidParameters.new(:post) if post.blank?
+    raise Discourse::NotFound unless post.revisions.present?
+
+    updated_at = Time.zone.now
+    post.revisions.destroy_all
+
+    post.update(version: 1, public_version: 1, last_version_at: updated_at)
```

Comment:
I wonder if we need a transaction that wraps the `revisions.destroy_all` and `post.update` so that we're sure no change/edits were made to that post in between? 🤔

Reply:
I wonder if we also want to add an entry in the "staff logs" that staff member X purged all the revisions of post Y? 🤔

Reply:
See https://github.com/discourse/discourse/pull/19913/files#diff-aa9acf72eb98d44206ae613d9f4f115af425fffd477b6d49b535813008be1f0fR2103-R2118